pkg/cmd/uninstall.go

```diff
@@ -0,0 +1,224 @@
+
+	if !o.skipOperator {
+		if err = o.uninstallOperator(c); err != nil {
+			fmt.Print(err)
+			return
+		}
+		fmt.Printf("Camel-K Operator removed from namespace %s\n", o.Namespace)
+	}
+
+	return
+}
+
+func (o *uninstallCmdOptions) uninstallOperator(c client.Client) error {
+	api := c.AppsV1()
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=camel-k,camel.apache.org/component=operator",
+	}
+
+	deployments, err := api.Deployments(o.Namespace).List(listOptions)
+	if err != nil {
+		return err
+	}
+
+	for _, deployment := range deployments.Items {
+		api.Deployments(o.Namespace).Delete(deployment.Name, &metav1.DeleteOptions{})
+	}
+
+	return nil
+}
+
+func (o *uninstallCmdOptions) uninstallClusterWideResources(c client.Client) error {
+	if !o.skipCrd {
+		if err := o.uninstallCrd(c); err != nil {
+			return err
+		}
+		fmt.Printf("Camel-K Custom Resourse Definitions removed from namespace %s\n", o.Namespace)
+	}
+
+	if !o.skipRoleBindings {
+		if err := o.uninstallRoleBindings(c); err != nil {
+			return err
+		}
+		fmt.Printf("Camel-K Role Bindings removed from namespace %s\n", o.Namespace)
+	}
+
+	if !o.skipClusterRoles {
+		if err := o.uninstallClusterRoles(c); err != nil {
+			return err
+		}
+		fmt.Printf("Camel-K Cluster Roles removed from namespace %s\n", o.Namespace)
+	}
+
+	return nil
+}
+
+func (o *uninstallCmdOptions) uninstallCrd(c client.Client) error {
+	restClient, err := customclient.GetClientFor(c, "apiextensions.k8s.io", "v1beta1")
+	if err != nil {
+		return err
+	}
+
+	result := restClient.
+		Delete().
+		Param("labelSelector", "app=camel-k").
+		Resource("customresourcedefinitions").
+		Do()
+
+	if result.Error() != nil && !k8serrors.IsAlreadyExists(result.Error()) {
+		return result.Error()
+	}
+
+	return nil
+}
+
+func (o *uninstallCmdOptions) uninstallRoleBindings(c client.Client) error {
+	api := c.RbacV1()
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=camel-k",
+	}
+
+	roleBindings, err := api.RoleBindings(o.Namespace).List(listOptions)
+	if err != nil {
+		return err
+	}
+
+	for _, roleBinding := range roleBindings.Items {
+		api.ClusterRoles().Delete(roleBinding.Name, &metav1.DeleteOptions{})
+	}
+
+	return nil
+}
+
+func (o *uninstallCmdOptions) uninstallClusterRoles(c client.Client) error {
+	api := c.RbacV1()
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=camel-k",
+	}
+
+	clusterRoles, err := api.ClusterRoles().List(listOptions)
+	if err != nil {
+		return err
+	}
+
+	for _, clusterRole := range clusterRoles.Items {
+		api.ClusterRoles().Delete(clusterRole.Name, &metav1.DeleteOptions{})
+	}
+
+	return nil
+}
+
+func (o *uninstallCmdOptions) uninstallIntegrationPlatform(c client.Client) error {
+	api, err := customclient.GetDefaultDynamicClientFor("integrations", o.Namespace)
+	if err != nil {
+		return err
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=camel-k,camel.apache.org/component=operator",
```

Comment:
Isn't selecting on `app=camel-k` enough? I didn't think platforms were labelled with `camel.apache.org/component=operator`.

Reply:
this is correct, I have used 'app' only for filtering

Reply:
looks like the pr #884 need to be resolved for labeleing operator with app: camel-k